File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/application/book/command/command/RegisterBookCommand.java
package dev.starryeye.book_library.application.book.command.command;

import java.util.Objects;

public record RegisterBookCommand(
        String bookname
) {

    public RegisterBookCommand {
        Objects.requireNonNull(bookname, "bookname must not be null");
        if (bookname.isBlank()) {
            throw new IllegalArgumentException("bookname must not be blank");
        }
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/application/book/command/command/RegisterBookCommand.kt
package dev.starryeye.book_library.application.book.command.command

import dev.starryeye.book_library.domain.book.BookCategory

data class RegisterBookCommand(
    val bookname: String,
    val category: BookCategory,
) {

    init {
        require(bookname.isNotBlank()) { "bookname must not be blank" }
    }
}